src/com/xilinx/rapidwright/timing/TimingModel.java

```diff
@@ -500,17 +502,82 @@ protected List<TimingGroup> determineGroups(List<Node> nodes, List<IntentCode> n
         return result;
     }
 
+    private Pair<Integer,Integer> findReferenceTileLocation(Device d) {
+        class Helper {
+            Map<Integer, Set<Integer>> spans;
+            public Helper() {
+                spans = new HashMap<Integer,Set<Integer>>();
+            }
+            void insert(int val, int span) {
+                if (!spans.containsKey(span)) {
+                    spans.put(span, new HashSet<Integer>());
+                }
+                spans.get(span).add(val);
+            }
+            Set<Integer> longest() {
+                List<Integer> keys = new ArrayList<>(spans.keySet());
+                Integer maxSpan = Collections.max(keys);
+                return spans.get(maxSpan);
+            }
+        }
+
+
+        // for each column, look for valid row
+        Helper colHelper = new Helper();
+        for (int i = 0; i < d.getColumns(); i++) {
+            int span = 0;
+            for (int j = 0; j < d.getRows(); j++) {
+                if (d.getTile("INT_X" + i + "Y" + j) != null) {
+                    span++;
+                }
+            }
+            colHelper.insert(i,span);
+        }
+
+        // for each row, look for valid col
+        Helper rowHelper = new Helper();
+        for (int j = 0; j < d.getRows(); j++) {
+            int span = 0;
+            for (int i = 0; i < d.getColumns(); i++) {
+                if (d.getTile("INT_X" + i + "Y" + j) != null) {
+                    span++;
+                }
+            }
+            rowHelper.insert(j,span);
+        }
+
+        for (int INTCol : colHelper.longest()) {
+            for (int INTRow : rowHelper.longest()) {
+                Tile tile = d.getTile("INT_X" + INTCol + "Y" + INTRow);

```

Comment:
```suggestion
        for (int x : colHelper.longest()) {
            for (int y : rowHelper.longest()) {
                Tile tile = d.getTile("INT", x, y);

```
Variables should not start with capital letters and replacing tile getter with one that doesn't need to create String objects.